etl/steps/data/garden/un_sdg/2022-07-07/un_sdg.py

```diff
@@ -0,0 +1,360 @@
+import pandas as pd
+import numpy as np
+import json
+from structlog import get_logger
+from pathlib import Path
+from typing import Tuple, List, Any, Dict
+from owid.catalog.utils import underscore
+
+from owid.walden import Catalog
+from owid.catalog import Dataset, Table
+from etl.paths import DATA_DIR
+
+COUNTRY_MAPPING_PATH = Path(__file__).parent / "un_sdg.country_mapping.json"
+
+BASE_URL = "https://unstats.un.org/sdgapi"
+VERSION = Path(__file__).parent.stem
+FNAME = Path(__file__).stem
+NAMESPACE = Path(__file__).parent.parent.stem
+
+log = get_logger()
+
+
+def run(dest_dir: str, query: str = "") -> None:
+    log.info("Reading in dataset from Meadow...")
+    ds_meadow = Dataset((DATA_DIR / f"meadow/{NAMESPACE}/{VERSION}/{FNAME}").as_posix())
+
+    assert (
+        len(ds_meadow.table_names) == 1
+    ), "Expected meadow dataset to have only one table, but found > 1 table names."
+    tb_meadow = ds_meadow[FNAME]
+    df = pd.DataFrame(tb_meadow)
+    df = create_units(df)
+    df = manual_clean_data(df)
+
+    log.info("Harmonizing entity names...")
+    country_mapping = load_country_mapping()
+    excluded_countries = load_excluded_countries()
+    df = df[~df.country.isin(excluded_countries)]
+    assert df["country"].notnull().all()
+    countries = df["country"].map(country_mapping)
```

Comment:
@spoonerf I'm stealing your code and have noticed that you don't assign `countries` to dataframe. Shouldn't you call `df["country"] = countries` somewhere below?

Reply:
Very good point, that was me badly copying Bobbie's code - should be fixed now!